lib/Platform/Intl/PlatformIntlApple.mm

```diff
@@ -337,48 +493,1297 @@ LocaleMatch lookupMatcher(
+      u"segment",  u"shrd",     u"sind",     u"sinh",     u"sora",
+      u"sund",     u"takr",     u"talu",     u"tamldec",  u"telu",
+      u"thai",     u"tibt",     u"tirh",     u"vaii",     u"wara",
+      u"wcho"};
+  // Ref::
+  // https://github.com/unicode-org/cldr/blob/release-37/common/bcp47/collation.xml
+  // -- Minus "standard" & "search" which are not allowed as per spec:
+  // https://tc39.es/ecma402/#sec-intl-collator-internal-slots
+  static std::vector<std::u16string> coVector = {
+      u"big5han",
+      u"compat",
+      u"dict",
+      u"direct",
+      u"ducet",
+      u"emoji",
+      u"eor",
+      u"gb2312",
+      u"phonebk",
+      u"phonetic",
+      u"pinyin",
+      u"reformed",
+      u"searchjl",
+      u"stroke",
+      u"trad",
+      u"unihan",
+      u"zhuyin"};
+  static std::vector<std::u16string> caVector = {
+      u"buddhist",
+      u"chinese",
+      u"coptic",
+      u"dangi",
+      u"ethioaa",
+      u"ethiopic",
+      u"gregory",
+      u"hebrew",
+      u"indian",
+      u"islamic",
+      u"islamic-umalqura",
+      u"islamic-tbla",
+      u"islamic-civil",
+      u"islamic-rgsa",
+      u"iso8601",
+      u"japanese",
+      u"persian",
+      u"roc"};
+  auto pos = iter - keys.begin();
+  switch (pos) {
+    case 0:
+      return containsValidKeyword(nuVector, value);
+    case 1:
+      return containsValidKeyword(coVector, value);
+    case 2:
+      return containsValidKeyword(caVector, value);
+    default:
+      return false;
+  }
+};
+void setUnicodeExtensions(
+    const std::u16string &supportedExtendionKey,
+    const std::vector<std::u16string> &valueList) {
+  std::vector<std::u16string> unicodeExtensionKeywords;
+  // TODO: From BCP Parser
+  // https://github.com/facebook/hermes/blob/17d7583650e5f4b59645e0607e24c7ccb15404ee/lib/Platform/Intl/java/com/facebook/hermes/intl/LocaleObjectAndroid.java#L247
+};
+// https://tc39.es/ecma402/#sec-resolvelocale
+ResolveLocale resolveLocale(
+    const std::vector<std::u16string> &availableLocales,
+    const std::vector<std::u16string> &requestedLocales,
+    const Options &options,
+    const std::vector<std::u16string> &relevantExtensionKeys
+    //  In line with Java, haven't included LocaleData
+) {
+  //  Skip 1/2/3, as we don't need to have independant implementations for best
+  //  fit
+  auto localeMatchResult = lookupMatcher(requestedLocales, availableLocales);
+  //  5. Let result be a new Record.
+  ResolveLocale result;
+  std::u16string value;
+  std::vector<std::u16string> supportedExtensionAdditionKeys;
+  //  9. For each element key of relevantExtensionKeys, do
+  for (std::u16string key : relevantExtensionKeys) {
+    result.value = u"";
+    if (localeMatchResult.extension != u"") { // 9.h.
+      if (localeMatchResult.extension.find(key)) { // 9.h.i.
+        std::u16string requestedValue = localeMatchResult.extension;
+        if (requestedValue != u"") {
```

Comment:
requestedValue must be `keywords[key]`, and `keywords` must be defined in **7.**, which is missing. With current implementation this condition breaks. This check is essentially the one that we had in this line    ` if (localeMatchResult.extension != u"") { // 9.h.`, which means that `else` section is unreachable.
Please double check the correctness of this entire function.

Reply:
For JSC I noticed they used something like localeMatchResult.extension.split('-', keywords); to set the keywords instead of UnicodeExtensionComponents. Is there a built in function like split that could be used instead of writing out a loop for this?

Reply:
Unfortunately I can't think of any built-in C++ function that does that.